File: SpecialSumOfArray.java
/*
import java.util.* ;
import java.io.*; 
public class Solution{
	public static int specialSum(ArrayList<Integer> arr, int n ){
        // Write your code here.
		int sum = 0;
		for(int i : arr) {
			sum += i;
		}
		while(sum > 9) { //sum until it gets a one digit sum
			int sumOfSum = 0;
			while(sum > 0) {
				int rem = sum %10;
				sumOfSum += rem;
				sum /= 10;
			}
			sum = sumOfSum;
		}
		return sum;
	}
}
*/

//updated one

import java.util.* ;
import java.io.*; 
public class Solution{
	public static int specialSum(ArrayList<Integer> arr, int n ){
        // Write your code here.
		int sum = 0;
		for(int i : arr) {
			sum += i;
		}
		if(sum == 0) {
			return 0;
		}
		sum = sum%9;
		if(sum == 0) {
			return 9;
		}
		return sum;
	}
}

File: SpecialSumOfArray.cpp
#include <bits/stdc++.h> 
int specialSum(vector<int>& arr, int n){
    // Write your code here.
    // calculate the total sum
    int totalSum = 0;
    for(auto it : arr) totalSum += it;
    // we have to return the one digit sum
    // if the sum is 0, the return 0
    if(totalSum == 0) return 0;
    // which will be always between 0 - 9
    // we can take %9, totalSum = 0 case already considered
    // if the %9 == 0, then it's obvious the one digit sum is 0, so if it's 0 return 0,
    // else return the % value
    return (totalSum%9 == 0) ? 9 : totalSum%9;
}